src/Core/src/Platform/iOS/SwitchExtensions.cs

```diff
@@ -42,13 +45,10 @@ public static void UpdateThumbColor(this UISwitch uiSwitch, ISwitch view)
 
 		internal static UIView GetTrackSubview(this UISwitch uISwitch)
 		{
-			UIView uIView;
 			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsTvOSVersionAtLeast(13))
-				uIView = uISwitch.Subviews[0].Subviews[0];
+				return uISwitch.Subviews[0].Subviews[0];
 			else
-				uIView = uISwitch.Subviews[0].Subviews[0].Subviews[0];
-
-			return uIView;
+				return uISwitch.Subviews[0].Subviews[0].Subviews[0];
```

Comment:
This looks 🐟 , if I understand correctly you are trying to set the tint color, the UISwitch API already provides methods/properties for that:
* https://developer.apple.com/documentation/uikit/uiswitch/1623684-thumbtintcolor?language=objc
* https://developer.apple.com/documentation/uikit/uiswitch/1623687-ontintcolor?language=objc

AFAIK those properties are what your are trying to do and simplify the need for checking the version of the OS.

Reply:
Talking with @tj-devel709  there is no public API for what he wants and this is the only possible solution (we talked about a diff approach with layers) there is a possible crash with the way the code is written. If apple decides to change the tree view of the widget (go back to 2 for example) the code would be getting a NRE. Since accessing the tree is like accessing the private implementation of the widget the code should be written to protect MAUI from that (better to have a bug about colors than a NRE crash).

Reply:
What @mandel-macaque said - make this return `null` if the track isn't found, and null-check after line 20 above.